Source/sound.cpp

```diff
@@ -68,6 +71,16 @@ void CleanupMusic()
 #endif
 }
 
+std::vector<std::unique_ptr<SoundSample>> duplicateSounds;
+SoundSample *DuplicateSound(const SoundSample &sound) {
+	auto duplicate = std::make_unique<SoundSample>();
+	if (duplicate->DuplicateFrom(sound) != 0)

```

Comment:
Isn't there a nicer way to achieve this using copy constructors? Having a method that copies seems like a code smell to me, as it allows for the instance to be used in the meantime (and potentially used incorrectly).

This to me sounds like temporal coupling, similar to having "Initialize" methods.

Reply:
Good idea, will do

Reply:
Actually, this would require quite a bit of refactoring because the construction can fail (meaning we can't just use a copy constructor -- as we don't use exceptions). Leaving for later.